### PR TITLE
WFLY-17164 Remove AttributeDefinition.isAllowNull

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFLY-17164

### CODE REVIEW

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/PropertiesAttributeDefinition.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ public Builder(PropertiesAttributeDefinition basis) {
         @Override
         public PropertiesAttributeDefinition build() {
             if (this.elementValidator == null) {
-                this.elementValidator = new ModelTypeValidator(ModelType.STRING, this.isAllowNull(), this.isAllowExpression());
+                this.elementValidator = new ModelTypeValidator(ModelType.STRING, this.isNillable(), this.isAllowExpression());
```
Comment (pferraro) [inline clustering/common/src/main/java/org/jboss/as/clustering/controller/PropertiesAttributeDefinition.java:135]: isNillable() is not a method of MapAttributeDefinition.Builder.

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/validation/AbstractParameterValidatorBuilder.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public ParameterValidatorBuilder configure(AttributeDefinition definition) {
     @Override
     public ParameterValidatorBuilder configure(AbstractAttributeDefinitionBuilder<?, ?> builder) {
         this.allowsExpressions = builder.isAllowExpression();
-        this.allowsUndefined = builder.isAllowNull();
+        this.allowsUndefined = builder.isNillable();
```
Comment (pferraro) [inline clustering/common/src/main/java/org/jboss/as/clustering/controller/validation/AbstractParameterValidatorBuilder.java:46]: isNillable() is not a method of AbstractAttributeDefinitionBuilder.

**Review by pferraro — CHANGES_REQUESTED** [review]

**Comment by yersan** [comment]
Hi @parsharma , for your information,  we have already released the WildFly Core version required to complete this PR.

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/PropertiesAttributeDefinition.java (hunk 1 above)
Comment (pferraro) [reply clustering/common/src/main/java/org/jboss/as/clustering/controller/PropertiesAttributeDefinition.java:135]: OK - this compiles now.  I did not realize that this PR was dependent on a future wildfly-core release.

**Comment by pferraro** [comment]
@parsharma I think you are going to need to rebase your branch in order to get the gihub actions to pass (unlike the CI jobs, these are run directly on your branch).

**Comment by parsharma** [comment]
Thanks for letting me know and I am sorry I was on sick leave last week so won't be able to rebase the branch.

**Review by pferraro — APPROVED** [review]

**Comment by pferraro** [comment]
Thanks @parsharma !

**Pull request merged**